### DISCUSSION TITLE
Publishing profiling attribute names in flagged output

### DISCUSSION THREAD

**Discussion by stefan-jung** [dita-ot/dita-ot#4466] [Q&A] [answered]
Let's say I have a topic like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE topic PUBLIC "-//OASIS//DTD DITA Topic//EN" "topic.dtd">
<topic id="topic">
    <title>topic</title>
    <body>
        <p props="A">A</p>
        <p props="A B">A B</p>
        <p props="A B C">A B C</p>
    </body>
</topic>
```

And a DITAVAL file like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<val>
    <prop action="flag" att="props" val="A" color="red"/>
    <prop action="flag" att="props" val="B" color="green"/>
    <prop action="flag" att="props" val="C" color="blue"/>
</val>
```

I get a rendered output of this:

![image](https://github.com/dita-ot/dita-ot/assets/3266567/91230cb6-d8f5-4877-8107-9aa9e74cfe9d)

The colors are nice, but they do not help me when the filtered element has more than just one profiling attribute value. In this particular case, they could be even misleading. Oxygen XML handles the problem quite nicely, because it renders the profiling attribute names and values in a CSS:after.

![image](https://github.com/dita-ot/dita-ot/assets/3266567/917c6d23-c1ea-4a28-95e5-95af18595a5a)

But is it possible to do the same in PDF (XSL:FO FOP)? This is probably a very common problem, so maybe someone can share some code or a plugin.

Test files: [flag-me.zip](https://github.com/dita-ot/dita-ot/files/15067414/flag-me.zip)

**Comment by raducoravu** [comment, marked as answer]
Maybe instead of action=flag you could use action=passthrough so that the profiling attributes make it through to the DITA topics in the temporary folder and then you could have some custom XSLT templates which create some XSL-FO content for each element which has profiling attributes on it.
